luavm/src/lib/lua_values/mod.rs

```diff
@@ -57,7 +57,16 @@ impl LuaVal {
                 // The behaviour of `as f64` is the same as the conversion
                 // from int to float described in the manual.
                 LuaValKind::INT => Ok(((self.val >> tagging::TAG_SHIFT) as i64) as f64),
-                LuaValKind::FLOAT => Ok(transmute::<usize, f64>(self.val ^ self.kind() as usize)),
+                LuaValKind::FLOAT => {
+                    // in the case of floats, the lower bits contain the tag, thus
```

Comment:
Now that we're using `LuaValKind::FLOAT`, I think we can remove the comment: it's now much more obvious to me (personally) what's going on.